test/e2e/admin_portal.go

```diff
@@ -0,0 +1,187 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"                           //nolint
+	. "github.com/onsi/gomega"                           //nolint
+	. "github.com/serge1peshcoff/selenium-go-conditions" //nolint
+	. "github.com/tebeka/selenium"                       //nolint
+)
+
+var _ = Describe("Admin Portal E2E Testing", func() {
+	BeforeEach(skipIfNotInDevelopmentEnv)
+	var wdPoint *WebDriver
+	var wd WebDriver
+	var host string
+	JustBeforeEach(func() {
+		host, wdPoint = adminPortalSessionSetup()
+		wd = *wdPoint
+		wd.Get(host + "/v2")
+		wd.Refresh()
+	})
+
+	AfterEach(func() {
+		if wd != nil {
+			wd.Quit()
+		}
+	})
+	It("Should be able to populate cluster data correctly", func() {
+		wd.Wait(ElementIsLocated(ByCSSSelector, "div[data-automation-key='name']"))
+
+		cluster, err := wd.FindElement(ByCSSSelector, "div[data-automation-key='name']")
+		if err != nil {
+			panic(err)
+		}
+
+		Expect(cluster.Text()).To(Equal(os.Getenv("CLUSTER")))
+	})
+
+	It("Should be able to filter cluster data correctly", func() {
+		wd.Wait(ElementIsLocated(ByCSSSelector, "div[data-automation-key='name']"))
+
+		filter, err := wd.FindElement(ByCSSSelector, "input[placeholder='Filter on resource ID']")
+		if err != nil {
+			panic(err)
+		}
+
+		// Set filter so it doesn't match cluster name
+		filter.SendKeys("Incorrect Cluster")
+
+		wd.Wait(ElementIsLocated(ByCSSSelector, "span.itemsCount-162"))
+		text, err := wd.FindElement(ByCSSSelector, "span.itemsCount-162")
+		if err != nil {
+			panic(err)
+		}
+
+		Expect(text.Text()).To(Equal("Showing 0 items"))
+	})
+
+	It("Should be able to populate cluster info panel correctly", func() {
+		wd.Wait(ElementIsLocated(ByCSSSelector, "div[data-automation-key='name']"))
+
+		cluster, err := wd.FindElement(ByCSSSelector, "div[data-automation-key='name']")
+		if err != nil {
+			panic(err)
+		}
+
+		cluster.Click()
+
+		wd.Wait(ElementIsLocated(ByCSSSelector, "ms-Panel is-open ms-Panel--hasCloseButton ms-Panel--custom root-225"))
+
+		panelFields, err := wd.FindElements(ByCSSSelector, "span.css-287")
+		if err != nil {
+			panic(err)
+		}
+		var filteredPanelFields []string
+		for _, panelField := range panelFields {
+			panelText, err := panelField.Text()
+			if err != nil {
+				panic(err)
+			}
+
+			if panelText != ":" {
+				filteredPanelFields = append(filteredPanelFields, panelText)
+			}
+		}
+
+		Expect(panelFields).ShouldNot(BeEmpty())
```

Comment:
This expectation failed in the pipeline: https://msazure.visualstudio.com/AzureRedHatOpenShift/_build/results?buildId=56020765&view=logs&j=3965c7ea-79de-5618-9d9a-42b627bfb354&t=4810cc2a-45e4-5107-9a11-ae0073a144ef&l=1431

Reply:
Finally resolved!